src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -95651,6 +95648,126 @@ class Box<T>
                 Diagnostic(ErrorCode.ERR_BadBinaryOps, "x == null").WithArguments("==", "T", "<null>").WithLocation(53, 13));
         }
 
+        [Fact, WorkItem(34942, "https://github.com/dotnet/roslyn/issues/34942")]
+        public void ConditionalAccess_01()
+        {
+            var source =
+@"#nullable enable
+using System;
+
+public class Program
+{
+    static void Main()
+    {
+        string? value = ""42"";
+        int count = 84;
+        if (value?.Length == count)
+        {
+            Console.WriteLine(value.Length);
+        }
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(34942, "https://github.com/dotnet/roslyn/issues/34942")]
+        public void ConditionalAccess_02()
+        {
+            var source =
+@"#nullable enable
+using System;
+
+public class Program
+{
+    static void Main()
+    {
+        string? value = ""42"";
+        if (value?.Length == (int?) null)
+        {
+            Console.WriteLine(value.Length);
+        }
+        else
+        {
+            Console.WriteLine(value.Length);
+        }
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                    // (11,31): warning CS8602: Dereference of a possibly null reference.
+                    //             Console.WriteLine(value.Length);
+                    Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "value").WithLocation(11, 31));
+        }
+
+        [Fact, WorkItem(34942, "https://github.com/dotnet/roslyn/issues/34942")]
+        public void ConditionalAccess_03()
+        {
+            var source =
+@"#nullable enable
+using System;
+
+public class Program
+{
+    static void Main()
+    {
+        string? value = ""42"";
+        int? i = null;
+        if (value?.Length == i)
+        {
+            Console.WriteLine(value.Length);
```

Comment:
Please add some `// 1|2` comments to these so we can tell them apart easily.

Reply:
Done